### PR TITLE
[FIX] mrp: assign lots on by-products

### PR DESCRIPTION
**Pull request by adwid**
When trying to assign a lot on a "by-product", an error is raised.

To reproduce the issue:
1. In Settings, enable "By-Products"
2. Create three products P_compo, P_finish01, P_finish02
    - P_finish02 is tracked
3. Create a bill of materials:
    - Product: P_finish01
    - Type: Manufacture
    - Components: 1 x P_compo
    - By-products: 1 x P_finish02
4. Create and confirm a MO with 1 x P_finish01
5. Open the detailed operations of P_finish02 and try to add a lot on
the line

Error: An Odoo Client Error is raised "Error: KeyError:
'raw_material_production_id'"

When clicking on the detailed operations, we open the form view of the
related stock move. This view is based on
`stock.view_stock_move_operations` and there are actually three
primary-type inheritances of it:
- V1: `mrp.view_stock_move_operations_raw`
    - It gets the field `raw_material_production_id`
    - It is used on a MO, when opening the detailed operations of a
component
- V2: `mrp.view_stock_move_operations_finished`
    - It does not get the field `raw_material_production_id`
    - It is used on a MO, when opening the detailed operations of a
finished move (for instance step 5 in above use case)
- V3: `stock.view_stock_move_nosuggest_operations`
    - It does not get the field `raw_material_production_id`
    - It is used on a picking, when opening the detailed operations of a
transferred product

As said, all these three views are based on
`stock.view_stock_move_operations`. This view lists all related stock
move lines thanks to another view:
`stock.view_stock_move_line_operation_tree`
https://github.com/odoo/odoo/blob/289d0d845df092eeae6ce0598ffef9158978ad30/addons/stock/views/stock_move_views.xml#L177
And in MRP, this tree view is extended by
`mrp.view_stock_move_line_operation_tree_finished`. This extension adds
the value of `raw_material_production_id` in the context of the field
`lot_id`:
https://github.com/odoo/odoo/blob/6d0ddc388d1c389e1911a6bb1c59c252870da1a7/addons/mrp/views/stock_move_views.xml#L46-L53
(this is used later to check if, on a component, the new lot can be
created. See [1] for more details).

In conclusion, this structure can not work: we have three views based on
the form `stock.view_stock_move_operations`. This form view lists all
SML thanks to a tree view which is overridden and uses the field
`raw_material_production_id`. However, only V1 will get and provide a
value for that field. So, when using V2 or V3 (basic use case: deliver a
tracked product), it will raise an error.

[1] 6d0ddc388d1c389e1911a6bb1c59c252870da1a7

OPW-2942049

### CODE REVIEW

**Comment by adwid** [comment]
Fixed by #97071
